File: WEEK20/그리디_큰 수 만들기.js
function solution(number, k) {
  let stack = new Array();
  let count = 0;

  for (let i = 0; i < number.length; i++) {
    const num = number[i];

    if (count === k) {
      stack.push(num);
      continue;
    }

    if (stack.length === 0) {
      stack.push(num);
    } else {
      if (stack[stack.length - 1] >= num) {
        stack.push(num);
      } else {
        while (stack[stack.length - 1] < num) {
          count += 1;
          stack.pop();

          if (count === k) break;
          if (stack.length === 0) break;
        }
        stack.push(num);
      }
    }
    if (count !== k) {
      answer = stack.slice(0, number.length - k).join("");
    }
  }
  answer = stack.join("");

  return answer;
}

File: WEEK20/그리디_큰 수 만들기.py
def solution(number, k):
    answer = ''
    stack = []
    count = 0
    
    for num in number:
        if count == k:
            stack.append(num)
            continue
        
        if len(stack) == 0:
            stack.append(num)
        else:
            if stack[len(stack) - 1] >= num:
                stack.append(num)
            else:
                while stack[len(stack) - 1] < num:
                    count += 1
                    stack.pop()
                    
                    if count == k:
                        break
                    if len(stack) == 0:
                        break
                stack.append(num)
    
    if count != k:
        answer = ''.join(stack[:len(number) - k])
        return answer
    
    answer = ''.join(stack)
    return answer

